yt_dlp/extractor/nrk.py

```diff
@@ -163,33 +171,36 @@ def call_playback_api(item, query=None):
         playable = manifest['playable']
 
         formats = []
+        subtitles = {}
         for asset in playable['assets']:
             if not isinstance(asset, dict):
                 continue
             if asset.get('encrypted'):
-                continue
+                pass  # Unencrypted stream no longer available
             format_url = url_or_none(asset.get('url'))
             if not format_url:
                 continue
             asset_format = (asset.get('format') or '').lower()
             if asset_format == 'hls' or determine_ext(format_url) == 'm3u8':
-                formats.extend(self._extract_nrk_formats(format_url, video_id))
+                fmts, subs = self._extract_nrk_formats_and_subtitles(format_url, video_id)
+                formats.extend(fmts)
+                self._merge_subtitles(subs, target=subtitles)
             elif asset_format == 'mp3':
                 formats.append({
                     'url': format_url,
                     'format_id': asset_format,
                     'vcodec': 'none',
                 })
 
-        data = call_playback_api('metadata')
+        data = self._call_api(try_get(manifest, lambda x: x['_links']['metadata']['href']), video_id, 'metadata')
 
-        preplay = data['preplay']
-        titles = preplay['titles']
-        title = titles['title']
-        alt_title = titles.get('subtitle')
+        preplay = try_get(data, lambda x: x['preplay'])
+        titles = try_get(preplay, lambda x: x['titles'])
+        title = try_get(titles, lambda x: x['title'])
+        alt_title = try_get(titles, lambda x: x['subtitle'])
 
         description = try_get(preplay, lambda x: x['description'].replace('\r', '\n'))
-        duration = parse_duration(playable.get('duration')) or parse_duration(data.get('duration'))
+        duration = parse_duration(playable.get('duration')) or parse_duration(data.get('duration')) or self._extract_m3u8_vod_duration(formats[0]['url'], video_id)
```

Comment:
This was added because the duration extracted from the metadata on `nrk.no/skole` videos did not match previous tests or the duration of the actual file. It should only do the additional m3u8 request if it is not able to find the duration in the playback or metadata objects, which then should only happen when it's coming from the `NRKSkoleIE` extractor since I didn't extract the duration from there.